docassemble/ALToolbox/al_income.py

```diff
@@ -661,20 +673,58 @@ class ALItemizedValue(DAObject):
         when calculating totals.
     """
 
+    def fields(self, use_exists=True) -> List[Dict[str, Any]]:
+        if use_exists:
+            return [
+                {
+                    "label": self.display_name,
+                    "field": self.attr_name("exists"),
+                    "datatype": "yesno",
+                },
+                {
+                    "label": word("Amount"),
+                    "field": self.attr_name("value"),
+                    "show if": self.attr_name("exists"),
+                    "datatype": "currency",
+                    "min": 0,
+                },
+            ]
+        else:
+            return [
+                {
+                    "label": self.display_name,
+                    "field": self.attr_name("value"),
+                    "datatype": "currency",
+                    "min": 0,
+                },
+            ]
+
     def total(self) -> Decimal:
         # If an item's value doesn't exist, use a value of 0
         # TODO: is this behavior correct, or should it force gathering the value?
         # What does a no-value item in the list represent?
-        if not hasattr(self, "value"):
+        if not hasattr(self, "value") or hasattr(self, "exists") and not self.exists:
             return Decimal(0)
 
         return _currency_float_to_decimal(self.value)
 
-    def __str__(self):
+    def __str__(self) -> str:
         """Returns a string of the value of the item with two decimal places."""
         currency_str = "{:.2f}".format(self.value)
         return currency_str
 
+    def __float__(self) -> float:
+        if hasattr(self, "exists") and not self.exists:
+            return 0.0
+        else:
+            return float(self.value)
```

Comment:
Yeah we had debated adding this before but I think you're right to do so.

Reply:
Really was necessary for the attachment block